doc/lua_api.txt

```diff
@@ -5959,20 +5993,87 @@ object you are working with still exists.
+    * `vel_x` is a number (default: `0.0`)
+    * `vel_z` is a number (default: `0.0`)
+    * If the object's velocity is fixed by `lock_velocity()` or `set_speed()`,
+      then it will remain unchanged.
 * `add_velocity(vel)`
     * `vel` is a vector, e.g. `{x=0.0, y=2.3, z=1.0}`
     * In comparison to using get_velocity, adding the velocity and then using
-      set_velocity, add_velocity is supposed to avoid synchronization problems.
+      set_velocity, this method is supposed to avoid synchronization problems.
 * `get_velocity()`: returns the velocity, a vector
+* `lock_velocity()`: fixes the velocity along the X and Z axes
+    * Can prove useful to maintain a specific heading after encountering an
+      obstacle, rather than continuously calling `set_velocity()`.
+    * Note that this will disable acceleration along the X and Z axes.
+* `unlock_velocity()`: frees the velocity along the X and Z axes
+* `is_velocity_locked()`: returns true if the velocity has been fixed
+* `set_speed(speed, yaw_offset)`
+    * Automatically fixes the velocity along the X and Z axes to correspond 
+      with the current heading (yaw).
+    * `speed` is a number equal to the length of the vector (vel_x,vel_z).
+    * `yaw_offset` is a number representing the angular offset from normal in 
+      radians, where `0` is forward and `pi` is backward (default: `0`).
+    * The object's normal is determined by its `yaw_origin` property.
+    * Player-like side-stepping can be achieved with either of the following 
+      values for `yaw_offset`:
+        * `pi / 2`: strafe left
+        * `-pi / 2`: strafe right
+    * A negative value for `speed` may also be used for back-tracking.
+* `add_speed(speed): increases or decreases the speed
+    * `speed` is a number
+* `set_speed_lateral(speed_x, speed_y)`
+    * Similar to `set_speed()` but calculates the angular offset from normal
+      given the magnitude of speed as a 2-dimensional vector (x,y) within the
+      object's local reference frame.
+    * `speed_x` is a number representing lateral motion, where negative moves
+      left, positive moves right, and zero stops (default: `0`)
+    * `speed_y` is a number, where positive moves forward, negative moves
+      backward, and zero stops (default: `0`)
+    * If `speed_x` is zero, then this method is equivalent to `set_speed()`
+      with `nil` for the second parameter. Other equivalencies include
+        * `set_speed_lateral(0, speed)` -> `set_speed(speed)`
+        * `set_speed_lateral(0, -speed)` -> `set_speed(-speed)`
+        * `set_speed_lateral(0, -speed)` -> `set_speed(speed, pi)`
+        * `set_speed_lateral(-speed)` -> `set_speed(speed, pi / 2)`
+        * `set_speed_lateral(speed)` -> `set_speed(speed, -pi / 2)`
+    * If the model is not oriented correctly, it may be necessary to set the
+      object's `yaw_origin` property to correspond with its normal.
 * `set_acceleration(acc)`
     * `acc` is a vector
+* `set_acceleration_vert(acc_y)`:
```

Comment:
Since it's rare to want to set acceleration horizontally, exclusive of the Y component, that is why I didn't include it. What is really needed is a function to accelerate within a local reference frame, but that will have to wait for another PR.